apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FirestoreDb.cs

```diff
@@ -410,8 +420,12 @@ public async Task<T> RunTransactionAsync<T>(Func<Transaction, Task<T>> callback,
                     }
                     catch (RpcException e) when (CheckRetry(e, ref rollback))
                     {
-                        // On to the next iteration...
+                        // On to the next iteration after a backoff.
                     }
+
+                    // This is essentially the inner loop of RetryAttempt.CreateRetrySequence.
+                    await scheduler.Delay(timing.BackoffJitter.GetDelay(backoff), cancellationToken).ConfigureAwait(false);
+                    backoff = timing.NextBackoff(backoff);
```

Comment:
I think this is clearer inside the catch.

Reply:
I'm generally nervous of doing stuff in catch blocks. There are some oddities about how code executes there - not that I can remember details. Let's chat next week.